core/src/main/groovy/noe/byteman/BytemanInstaller.groovy

```diff
@@ -0,0 +1,43 @@
+package noe.byteman
+
+import noe.common.utils.JBFile
+import noe.common.utils.PathHelper
+
+import java.nio.file.Files
+import java.nio.file.Paths
+
+/**
+ * Helping class for Byteman to handle copying byteman.jar file to visible location in the system or to user defined file.
+ */
+
+class BytemanInstaller {
+  protected static String DEFAULT_BYTEMAN_JAR_NAME = "byteman.jar"
+  protected static String DEFAULT_BYTEMAN_JAR_FOLDER = PathHelper.join(System.getProperty("java.io.tmpdir"), "noe", "byteman")
+
+  private File bytemanJar = null
+
+  BytemanInstaller() {
+    bytemanJar = new File(DEFAULT_BYTEMAN_JAR_FOLDER, DEFAULT_BYTEMAN_JAR_NAME)
```

Comment:
I'd recommend a groovy's default parameter here. That's essentially what you're doing--setting a default parameter if one is not provided. So instead of 2 constructors, we can have only one.

So, you could:
1.  create a new field: `private File defaultBytemanFile = new File(DEFAULT_BYTEMAN_JAR_FOLDER, DEFAULT_BYTEMAN_JAR_NAME)`
2. Use this as the default parameter of `targetBytemanJarFile`
3. Remove unnecessary constructor.

WDYT?

Reply:
Jobs finished. :)